cloudinit/config/cc_set_passwords.py

```diff
@@ -45,9 +45,11 @@ def get_users_by_type(users_list: list, pw_type: str) -> list:
     )
 
 
-def _restart_ssh_daemon(distro, service):
+def _restart_ssh_daemon(distro: Distro, service: str):
     try:
-        distro.manage_service("restart", service)
+        distro.manage_service(
+            "restart", service, "--job-mode='ignore-dependencies'"
```

Comment:
So something like this might work 
```
diff --git a/cloudinit/config/cc_set_passwords.py b/cloudinit/config/cc_set_passwords.py
index f58a1dba2..16074330f 100644
--- a/cloudinit/config/cc_set_passwords.py
+++ b/cloudinit/config/cc_set_passwords.py
@@ -45,10 +45,10 @@ def get_users_by_type(users_list: list, pw_type: str) -> list:
     )
 
 
-def _restart_ssh_daemon(distro: Distro, service: str):
+def _restart_ssh_daemon(distro: Distro, service: str, *extra_args: str):
     try:
         distro.manage_service(
-            "restart", service, "--job-mode=ignore-dependencies"
+            "restart", service, extra_args
         )
         LOG.debug("Restarted the SSH daemon.")
     except subp.ProcessExecutionError as e:
@@ -118,7 +118,7 @@ def handle_ssh_pwauth(pw_auth, distro: Distro):
             # for backwards compatibility so that users who think that they
             # need to manually start cloud-init (why?) with systemd (again,
             # why?) can do so.
-            _restart_ssh_daemon(distro, service)
+            _restart_ssh_daemon(distro, service, "--job-mode=ignore-dependencies")
     else:
         _restart_ssh_daemon(distro, service)
```